pkg/services/ngalert/api/api_ruler.go

```diff
@@ -652,15 +679,21 @@ func (srv RulerSrv) getAuthorizedRuleGroup(ctx context.Context, c *contextmodel.
 	return rules, nil
 }
 
+type authorizedRuleGroupQuery struct {
```

Comment:
Fair point on the struct, it could go either way. I chose to go with a struct (and the "query" suffix) because this is a "search". No strong preference on moving it back to params if we feel this doesn't fit.

Reply:
IMO, it's fine with the struct. I can understand calling it a `Query` since it's a wrapper around a db call. Maybe the issue is that this should instead be a `store.ListAlertRules` call followed by a call to something like `srv.filterAuthorized{Rules,Groups}`.